kotlin/codegen/src/main/java/com/squareup/moshi/kotlin/codegen/metadata.kt

```diff
@@ -87,7 +93,14 @@ internal fun primaryConstructor(kotlinApi: TypeSpec, elements: Elements): Target
     )
   }
 
-  return TargetConstructor(parameters, primaryConstructor.modifiers.visibility())
+  val kmConstructorSignature = primaryConstructor.tag<ImmutableKmConstructor>()?.signature?.toString()
+      ?: run {
+        messager.printMessage(ERROR, "No KmConstructor found for primary constructor.",
```

Comment:
Is this something that is expected to happen (in cases of misconfiguration, or something)? If so, is there a better error message that users might understand?  I'm not sure this would be decipherable by an end user, or tell them how to fix the problem.

Reply:
I see below that they'll also receive "No primary constructor found on $element", which seems like a more understandable message.  Is this message required, then?  If it doesn't provide more specific information about how to solve the issue then I'd probably vote for leaving it off to avoid the issues filed referencing this because someone only looked at the first error message.

Reply:
Yeah I'm not sure what to do with these, because they're kind of things we know won't happen but don't want to just drop in a `!!` without something more descriptive. If it can gracefully skip it I've tried to do that, but fall back to a hard error that we'd recognize if they report it if it's something truly not something we can work with, like the `error()` case. Could add a request in a followup to file on Moshi if they see it?